vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -1431,84 +1400,71 @@ public void fetchCurrentUser(@NotNull ModelCallback<User> callback, @Nullable St
      * @param query         Query string for hitting the search endpoint
      * @param refinementMap Used to refine lists (generally for search) with sorts and filters
      * @param fieldFilter   The string of fields to include in the response (highly recommended!)
-     *                      {@link SearchRefinementBuilder}
+     *                      {@link RequestRefinementBuilder}
+     * @param caller        The {@link Caller} for the expected response type
      * @see <a href="https://developer.vimeo.com/api/spec#common-parameters">Vimeo API Field Filter Docs</a>
      */
     @Nullable
-    public Call<Object> fetchContent(@NotNull String uri, CacheControl cacheControl,
-                                     @NotNull ModelCallback callback, @Nullable String query,
-                                     @Nullable Map<String, String> refinementMap,
-                                     @Nullable String fieldFilter) {
+    public <DataType_T> Call<DataType_T> getContent(@NotNull String uri,
```

Comment:
I can't really think of a way around this - but as it stands now, you need to declare your type in the VimeoCallback and also with the caller (on top of the fact that both are necessary as of right now).

It would be cool if you only needed one of these - and you could figure out which caller was needed based on the type that was supplied to the callback. I think we've talked about passing the class in and switching off that, but that had its problems. It just seems like potentially avoidable boilerplate since those two types will always have to be the same, and we only have one caller object per type.

Not sure if you have any other thoughts on this (pros/cons of Caller being it's own argument, alternatives you've thought of).